### ISSUE TITLE
Obsidian freezes and must be killed in certain conditions (easy to reproduce)

### ISSUE THREAD

**Issue by Fred-Vatin** [FlorianWoelki/obsidian-iconize#574]
Obsidian| OS                           | Iconize
------- | ---------------------------- | -------
1.6.7   | Win 10 x64 pro 22H2          | v2.14.3

**To Reproduce**
1. Open the sandbox vault
2. Install Iconize
3. **IMPORTANT** : enable twemoji as emoji style and don’t touche the other options. All is default.
4. Restart Obsidian
5. Create a new note and insert this unicode char `™`
6. Press `CTRL+E` to switch to reading view
7. Obsidian freezes and you must kill it and relaunch

**Related issues**
#524

**Comment by spiritualgeek** [comment]
Also, constant freezing when "Toggle icon in links" is switched on. Along with #566, iconize has become completely unusable

**Comment by FlorianWoelki** [comment]
> Also, constant freezing when "Toggle icon in links" is switched on. Along with #566, iconize has become completely unusable 

I don't think that #566 is related to this issue. The official version of Obsidian >1.7 is not released yet. Therefore, I am unable to patch anything.

**Comment by spiritualgeek** [comment]
No worries, In my quest for minimalism I've decided to trim back on plugin use. Thanks

**Comment by tustin2121** [comment]
This is a particularly annoying bug for me. I'm implementing this workaround, so the app at least doesn't freeze forever in this callback:

```diff
--- a/.obsidian/plugins/obsidian-icon-folder/main.js
+++ b/.obsidian/plugins/obsidian-icon-folder/main.js
@@ -5495,6 +5495,7 @@ const createTreeWalker = (plugin, root) => {
 const checkForTextNodes = (treeWalker, match, cb) => {
     let currentNode = treeWalker.currentNode;
     while (currentNode) {
+        if (currentNode.textContent == "™") break; // #574
         if (currentNode.nodeType === Node.TEXT_NODE) {
             const text = currentNode;
             const textNodes = [...Array.from(text.parentElement.childNodes)].filter((n) => n instanceof Text);
```

It seems like `treewalker.nextNode()` is returning the current node when the TM unicode character is used, so that the while loop condition never becomes false. When I try to run `treewalker.nextNode()` from the inspector while breakpointed in the function, I got some exception about the callback no longer being valid, so idk what's going on there.

**Comment by chiefnoah** [comment]
This is still present and works with multiple unicode special symbols such as ™ and ©. I suspect either Iconize or Obsidian itself is doing some sort of conversion of the text automatically to map `:copyright:` / `:trademark:` to their corresponding code points and vice-versa. When opening the file in another editor, neovim in my case, the unicode symbol is present in the underlying file.
